modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -125,6 +127,9 @@
   @Inject
   private ClusterService clusterService;
 
+  @Inject
+  private PayloadStatisticsCursorDecoratorFactory payloadStatisticsCursorDecoratorFactory;
```

Comment:
Maybe use an interface in this case. Just in case you want to override the default statistic cursor decorator in the registry for special envs?